src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -5,10 +5,113 @@
 
 'use strict'
 
+import * as child_process from 'child_process'
 import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
 import { fileExists } from '../../filesystemUtilities'
-import { DefaultSamCliTaskInvoker } from './samCliInvoker'
-import { SamCliTaskInvoker } from './samCliInvokerUtils'
+import { ChildProcess } from '../../utilities/childProcess'
+import { ChannelLogger } from '../../utilities/vsCodeUtils'
+
+const localize = nls.loadMessageBundle()
+
+export const WAIT_FOR_DEBUGGER_MESSAGES = {
+    PYTHON: 'Waiting for debugger to attach...',
+    NODEJS810: 'Debugger listening on',
+}
+
+export interface SamLocalInvokeCommandArgs {
+    command: string,
+    args: string[],
+    options?: child_process.SpawnOptions,
+    waitForDebuggerAttachMessage: boolean,
+}
+
+/**
+ * Represents and manages the SAM CLI command that is run to locally invoke SAM Applications.
+ */
+export interface SamLocalInvokeCommand {
+    invoke({ }: SamLocalInvokeCommandArgs): Promise<void>
```

Comment:
Out of curiosity, is there any particular reason you wrote the param this way instead of `args: SamLocalInvokeCommandArgs`?

Reply:
No, it was more so that future params could go there if necessary.